### PR TITLE
[payment-endpoint] GitHub: fix reconcilePlan

### PR DESCRIPTION
**Pull request by geropl**
## Description
This PR fixes two issues that in combination caused https://github.com/gitpod-io/ops/issues/339:
 - [0b72f48](https://github.com/gitpod-io/gitpod/commit/0b72f4851dba42fda8c6927af3a3142fa6388976): reference `PendingGitHubEvents` by id and not by full shape for deletion: This ensures typeorm does not choke on unexpected fields ([identity](https://console.cloud.google.com/logs/query;cursorTimestamp=2021-11-25T14:05:29.182Z;query=resource.type%3D%22k8s_container%22%0Aresource.labels.pod_name:%22payment-endpoint-%22%0Atimestamp%3D%222021-11-25T14:05:26.842Z%22%0AinsertId%3D%22h89m9xljfvzh70c6%22?project=gitpod-191109))
 - [8bbb5fc](https://github.com/gitpod-io/gitpod/commit/8bbb5fc4117e3766f61bba8cc6a3a473ec601b23) and [ede2412](https://github.com/gitpod-io/gitpod/commit/ede241292cae98807fbed920c2b2a159cc26081f): Ensure that event reconciliation and bookkeeping are running inside the same transaction. This required a bit more code in `gitpod-db` as we so far had no interface to do so. The current solution is not perfect (too much code per DB) but IMO a reasonable compromise.

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/339

## How to test
I'm not sure how to test this in devstaging :confused: 

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#339*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ee/payment-endpoint/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/payment-endpoint/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by geropl** [comment]
/werft run

To make GitHub pick up the latest state from werft

File: components/ee/payment-endpoint/src/github/subscription-reconciler.ts [hunk 1]
```diff
@@ -219,27 +212,30 @@ export class GithubSubscriptionReconciler {
...
-            const subscriptions = await db.findAllSubscriptionsForUser(userId);
-            const userPaidSubscriptions = subscriptions.filter(s => UserPaidSubscription.is(s) && s.paymentReference.startsWith("github:"));
-
-            const model = new SubscriptionModel(userId, userPaidSubscriptions);
-            const success = await this.subscriptionMapper.map(evt, model);
-            if (!success) {
-                log.debug({userId}, "subscription mapper did not succeed for GitHub market purchase event. See errors above.", { evt });
-                return;
-            }
+        const subscriptions = await this.accountingDB.findAllSubscriptionsForUser(userId);
```
Comment (JanKoehnlein) [inline components/ee/payment-endpoint/src/github/subscription-reconciler.ts:217]: There is a slight drift here: The read ops are not run in the same transaction as the writes as before, but I guess you're aware of that.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: c84319fbfb60e54a9cbd94bf42af93177093ab6e</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6901#issuecomment-979899163" title="LGTM">JanKoehnlein</a>*

Associated issue: *#339*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/ee/payment-endpoint/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/payment-endpoint/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/ee/payment-endpoint/src/github/subscription-reconciler.ts (hunk 1 above)
Comment (geropl) [reply components/ee/payment-endpoint/src/github/subscription-reconciler.ts:217]: > There is a slight drift here: The read ops are not run in the same transaction as the writes as before, but I guess you're aware of that.

Yes. That would only have implications if:
 - we had concurrent writers
 - rely on transaction-scoped locks to resolve those
 As we don't, this does not matter.

**Pull request merged**